waterdata/views.py

```diff
@@ -49,8 +49,11 @@ def monitoring_location(site_no):
                                                           'siteStatus': 'all',
                                                           'agencyCd': agency_cd})
         if parameter_data_resp.status_code == 200:
-            param_data = parse_rdb(parameter_data_resp.iter_lines(decode_unicode=True))
-            location_capabilities = get_capabilities(param_data)
+            param_data = [param_datum for param_datum in
+                          parse_rdb(parameter_data_resp.iter_lines(decode_unicode=True)) if param_datum['parm_cd']]
+            site_dataseries = [get_disambiguated_values(param_datum, app.config['NWIS_CODE_LOOKUP'], {}) for
+                               param_datum in param_data]
+            location_capabilities = set([param_datum['parm_cd'] for param_datum in param_data])
```

Comment:
set can take a generator, so you can avoid the intermediate list: `set(param_datum['parm_cd'] for param_datum in param_data)`

Reply:
Okay.